src/OpenApi/src/Services/Schemas/OpenApiSchemaStore.cs

```diff
@@ -27,6 +28,8 @@ internal sealed class OpenApiSchemaStore
         [new OpenApiSchemaKey(typeof(PipeReader), null)] = new JsonObject { ["type"] = "string", ["format"] = "binary" },
     };
 
+    private readonly ConcurrentDictionary<OpenApiSchema, string?> _schemasWithReference = new(OpenApiSchemaComparer.Instance);
```

Comment:
Can you say more about why this needs to be a `ConcurrentDictionary`?  Will readers and writers be operating at the same time?  Without having taken an overview of the code, it feels like it will be constructed once and then read-only forever.

Reply:
Yep, `CocurrentDictionary` isn't actually useful here or elsewhere. Fixed.